File: conteudos/insertion_sort/insertion_sort.py
def insertion_sort(arr):
    n = len(arr)
    for i in range(1, n):
        chave = arr[i]
        j = i - 1

        while j >= 0 and arr[j] > chave:
            arr[j + 1] = arr[j]
            j -= 1
        arr[j + 1] = chave
        
        
if __name__ == "__main__":
    numeros = [31, 41, 59, 26, 42, 58]
    insertion_sort(numeros)
    print(numeros)

File: conteudos/insertion_sort/InsertionSort.java
package insertion_sort;

import java.util.ArrayList;
import java.util.Arrays;

public class InsertionSort {
    public static void main(String[] args) {    
        Integer[] numerosArray = {5, 3, 8, 6, 2, 7};
        ArrayList<Integer> numeros = new ArrayList<>(Arrays.asList(numerosArray));

        sorted(numeros);

        System.out.println(numeros.toString());

    }

    private static void sorted(ArrayList<Integer> numeros) {
        int tamanho = numeros.size();

        for(int i=0; i<tamanho; i++) {
            int chave = numeros.get(i);
            int j = i -1;

            while (j >= 0 && numeros.get(j) > chave) {
                numeros.set(j+1, numeros.get(j));
                j--;
            }
            numeros.set(j+1, chave);
        }
    }
}
